x-pack/legacy/plugins/task_manager/task_store.ts

```diff
@@ -392,20 +426,32 @@ function paginatableSort(sort: any[] = []) {
   return [...sort, sortById];
 }
 
-function taskInstanceToAttributes(doc: TaskInstance): SavedObjectAttributes {
+const taskInstanceToAttributes = flow(
+  applyConcreteTaskInstanceDefaults,
+  serializeTaskInstanceFields
+);
+
+function serializeTaskInstanceFields(doc: ConcreteTaskInstance): SavedObjectAttributes {
   return {
     ...omit(doc, 'id', 'version'),
-    params: JSON.stringify(doc.params || {}),
-    state: JSON.stringify(doc.state || {}),
-    attempts: (doc as ConcreteTaskInstance).attempts || 0,
-    scheduledAt: (doc.scheduledAt || new Date()).toISOString(),
-    startedAt: (doc.startedAt && doc.startedAt.toISOString()) || null,
-    retryAt: (doc.retryAt && doc.retryAt.toISOString()) || null,
-    runAt: (doc.runAt || new Date()).toISOString(),
-    status: (doc as ConcreteTaskInstance).status || 'idle',
+    ...mapValues(pick(doc, isPlainObject), objectProp => JSON.stringify(objectProp)),
```

Comment:
Regarding the fragility - I thought we were stringifying because we cant store objects there - any idea why we're stringifying at all then?

regarding defaulting- it should be the same as before, but I separate them on purpose because setting defaults and serialising are two different things and it's harder to maintain code that does two unrelated things.
The reason I'm not inlining them is that we need `applyConcreteTaskInstanceDefaults` independently when we're omitting and merging data from partial updates (see the `reschedule` method).

As for `toISOString` - that's what we were already doing, so I kept it the same. I could investigate why we use it specifically, but either way changing it is probably for a separate PR, no?

Reply:
Oh, is that a SO restriction?  You can't have `object` type properties, with say an `enabled: false` (to prevent index explosion)?  If so, then ya, every object would need to be stringified. If it's not an SO restriction, seems like we shoulda used `object enabled=false` for these, but then that's water under the bridge, migrating is a possibility, for a future PR :-)

As mentioned, even though it may be fragile if we extend the SO later, I think we'd find out pretty quickly (ES wouldn't index the doc), so not a big deal.